src/view3d.cpp

```diff
@@ -250,11 +368,30 @@ void View3D::paintGL()
     QTime frameTimer;
     frameTimer.start();
 
-    m_incrementalFramebuffer->bind();
+    // Get window size
+#ifdef DISPLAZ_USE_QT4
+    double dPR = 1.0;
+#else
+    double dPR = devicePixelRatio();
+#endif
+    int w = width() * dPR;
+    int h = height() * dPR;
+
+    // detecting a change in the device pixel ratio, since only the new QWindow (Qt5) would
+    // provide a signal for screen changes and this is the easiest solution
```

Comment:
Yeah, I would have preferred to use the `screenChanged()` signal, but we would have to move away from `QMainWindow`, which seems like a bit of work.